app/assets/javascripts/components/projects/ProjectBoard.js

```diff
@@ -8,39 +8,111 @@ import Sprints from "../stories/Sprints";
 import History from "../stories/History";
 import { getColumns } from "../../selectors/columns";
 import * as Columns from '../../models/beta/column';
-import { createStory, closeHistory } from '../../actions/story';
+import { createStory, closeHistory, dragDropStory } from '../../actions/story';
 import AddStoryButton from '../story/AddStoryButton';
 import * as Story from 'libs/beta/constants';
 import PropTypes from 'prop-types';
 import { storyPropTypesShape } from '../../models/beta/story';
-import { projectBoardPropTypesShape } from '../../models/beta/projectBoard';
+import {
+  projectBoardPropTypesShape,
+  getNewPosition,
+  getNewSprints,
+  getNewState,
+  moveTask,
+  getBacklogStories,
+  getArray
+} from '../../models/beta/projectBoard';
 import Notifications from '../Notifications';
 import { removeNotification } from '../../actions/notifications';
 import StorySearch from '../search/StorySearch';
 import SearchResults from './../search/SearchResults';
 import ProjectLoading from './ProjectLoading';
+import { DragDropContext } from 'react-beautiful-dnd';
 
-export class ProjectBoard extends React.Component {
-  componentWillMount() {
-    this.props.fetchProjectBoard(this.props.projectId);
-  }
+const ProjectBoard = ({
+  projectId,
+  createStory,
+  closeHistory,
+  notifications,
+  removeNotification,
+  history,
+  projectBoard,
+  fetchProjectBoard,
+  dragDropStory,
+  chillyBinStories,
+  backlogSprints,
+  doneSprints,
+  fetchPastStories
+}) => {
+  const [newChillyBinStories, setNewChillyBinStories] = useState();
+  const [newBacklogSprints, setNewBacklogSprints] = useState();
+
+  useEffect(() => {
+    setNewBacklogSprints(backlogSprints);
+  }, [backlogSprints]);
+
+  useEffect(() => {
+    setNewChillyBinStories(chillyBinStories);
+  }, [chillyBinStories]);
+
+  useEffect(() => {
+    fetchProjectBoard(projectId);
+  }, []);
+
+  const onDragEnd = ({ source, destination, draggableId }) => {
+    const { sprintIndex: sprintDropIndex, columnId: dropColumn } = JSON.parse(destination.droppableId);
+    const { sprintIndex: sprintDragIndex, columnId: dragColumn } = JSON.parse(source.droppableId);
+    const isSameColumn = dragColumn === dropColumn;
+    const destinationArray = getArray(dropColumn, newBacklogSprints, newChillyBinStories, sprintDropIndex); // stories of destination column
+    const sourceArray = getArray(dragColumn, newBacklogSprints, newChillyBinStories, sprintDragIndex); // stories of source column
+    const dragStory = sourceArray[source.index];
+
+    if (!destination) {
+      return;
+    }
+
+    if (isSameColumn && source.index === destination.index) {
```

Comment:
`source.index` and `destination.index` has  been used in many places, consider add it into a `const`

Reply:
Yeah I agree.